src/sys/select.rs

```diff
@@ -307,11 +307,25 @@ where
     let readfds = readfds.map(|set| set as *mut _ as *mut libc::fd_set).unwrap_or(null_mut());
     let writefds = writefds.map(|set| set as *mut _ as *mut libc::fd_set).unwrap_or(null_mut());
     let errorfds = errorfds.map(|set| set as *mut _ as *mut libc::fd_set).unwrap_or(null_mut());
-    let timeout = timeout.map(|ts| ts.as_ref() as *const libc::timespec).unwrap_or(null());
     let sigmask = sigmask.map(|sm| sm.as_ref() as *const libc::sigset_t).unwrap_or(null());
 
-    let res = unsafe {
-        libc::pselect(nfds, readfds, writefds, errorfds, timeout, sigmask)
+    #[cfg(target_os = "nto")]
+    let res = match timeout {
+        Some(t) => {
+            let mut timeout = t.clone();
+            unsafe {libc::pselect(nfds, readfds, writefds, errorfds, timeout.as_mut() as *mut libc::timespec, sigmask)}
```

Comment:
This looks like a bug in Neutrino.  POSIX says that the timeout argument should be const.  Frankly, given how complicated this makes Nix's code, I don't think it's worth it for the benefit this provides.  I would rather just not bind pselect on this platform.

Reply:
As also described in my [previous comment](https://github.com/nix-rust/nix/pull/2055#discussion_r1279161236), there are API differences that I'm trying to work around as I cannot change the libc API. Would a small wrapper function help (see other comment)?

Reply:
We could make it work, but it would definitely ugly-up the code.  And this const-mut conversion is worse than the size widening of readlink.  Do we know that there are even any QNX users who want to use pselect?

Reply:
@AkhilTThomas what do you think about removing pselect or do you know if it is used?